lib/Conversion/TTIRToTTNN/TTIRToTTNN.cpp

```diff
@@ -186,6 +186,47 @@ class ReshapeOpConversionPattern : public OpConversionPattern<ttir::ReshapeOp> {
   }
 };
 
+class UnsqueezeOpConversionPattern
+    : public OpConversionPattern<ttir::UnsqueezeOp> {
+public:
+  using OpConversionPattern<ttir::UnsqueezeOp>::OpConversionPattern;
+
+  LogicalResult
+  matchAndRewrite(ttir::UnsqueezeOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    // Extract input tensor type
+    ::mlir::RankedTensorType inputType =
+        mlir::cast<::mlir::RankedTensorType>(adaptor.getInput().getType());
+
+    // Get the unsqueeze dimension
+    int32_t dim = adaptor.getDim();
+
+    // Get the shape of the input tensor
+    auto inputShape = inputType.getShape();
+    llvm::SmallVector<int32_t, 4> newShape;
```

Comment:
nit: Should we change to 5 elements? That's the max rank of the `newShape`.

Reply:
Sure